src/uct/ib/base/ib_device.h

```diff
@@ -266,6 +266,21 @@ ucs_status_t uct_ib_device_port_check(uct_ib_device_t *dev, uint8_t port_num,
                                       unsigned flags);
 
 
+/**
+ * Check if device has ECE capability
+ */
+ucs_status_t
+uct_ib_device_ece_check(uct_ib_device_t *dev,
+                        struct ibv_context *ctx, struct ibv_pd *pd);
```

Comment:
1. layout looks weird. pls run the code through clang-format.
2. why need to pass all these params? uct_ib_device_t should be enough.

Reply:
1. I'll use clang-format to check it.
2. @yosefe I'll remove ```struct ibv_context *ctx``` parameter here.  For ```struct ibv_pd *pd```, it can be reused, WDYT?

Reply:
can we move this function to ib_md.c and use uct_ib_md_t?